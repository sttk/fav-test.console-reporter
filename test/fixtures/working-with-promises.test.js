'use strict';

var Promise = (typeof Promise === 'function') ? Promise :
              require('promise-polyfill');
var assert = require('assert');
var should = require('should');

function DB() {
}

DB.prototype.clear = function() {
  return new Promise(function (cb) {
    cb();
  });
};

DB.prototype.save = function() {
  return new Promise(function (cb) {
    cb();
  });
};

DB.prototype.find = function() {
  return new Promise(function(resolve) {
    resolve(['Tobi', 'Loki', 'Jane']);
  });
};

var tobi = 'Tobi',
    loki = 'Loki',
    jane = 'Jane';

var db = new DB();

beforeEach(function() {
  return db.clear()
    .then(function() {
      return db.save([tobi, loki, jane]);
    });
});

describe('#find()', function() {
  it('respond with matching records', function() {
    return db.find({ type: 'User' }).should.eventually.have.length(3);
  });
  it('should complete this test', function(done) {
    return new Promise(function(resolve) {
      assert.ok(true);
      resolve();
    }).then(done);
  });
});

