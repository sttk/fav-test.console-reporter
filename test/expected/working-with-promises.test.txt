

  #find()
    âœ“ respond with matching records
    1) should complete this test


  1 passing (10ms)
  1 failing

  1) #find()
       should complete this test:
     Error: Resolution method is overspecified. Specify a callback *or* return a Promise; not both.




